### DISCUSSION TITLE
Remove a drive from mergerfs without losing files

### DISCUSSION THREAD

**Discussion by antonical** [trapexit/mergerfs#1075] [General]
I have a large mergerfs pool containing 5 x 4TB drives. I want to free one of those drives up for other use as I have enough free space across the 4 remaining drives.

What is the best way to remove it and ensure all its files are transferred to the other drives so no files are lost?

I couldn't find a tool to do this unless I have missed it in all my reading.

du -h on the directory for that drive shows 1.3TB of data used. 

I have approx 4TB of free space across the 4 drives.

Is it just a case of stopping the mergerfs mount. rsync the data to one of the drives and then balance after that?

Or remove the drive I want lice from mergerfs and rsync back the files from the removed drive to the mergerfs mount?

Any help appreciated.

Cheers
Tony

**Comment by trapexit** [comment]
I don't have a tool because it's a pretty straight forward process.

If you can afford to remove the drive while in use simply remove the branch from mergerfs and rsync the data to the pool.

If you don't want to have the data missing temporarily from the pool... create another pool with whatever settings relevant to your situation to spread files around and then rsync from the branch to the temporary pool.

**Reply by antonical** [reply]
Hey, thanks for the reply. 

Thinking further on it I don't think I can remove the drive 'live'. I think the safest way is probably to stop all shares and services using the mergerfs mount. 

Umount the mergerfs pool. 

Edit fstab mergerfs mount to remove the drive.

remount the mergerfs pool

rsync the data to the mergerfs pool

Then run mergerfs.balance to even the pool up.

Maybe a mergerfs.dedup to finish.

A bit long winded but I can't see a safer way.

Thanks
Tony

**Comment by trapexit** [comment]
The way I described before is perfectly safe.

Mount another pool minus the branch to remove and rsync to it. You don't need to balance. Just let mergerfs do its thing.  This is how I do it and it is safe. rsync writes a temp file and renames so it's atomic from the software perspective and the original pool will select one of the files afterwards just fine. 

You are welcome to do as you suggested but requires more work.

**Reply by antonical** [reply]
Went with the belt and braces approach!

As I had to shut down all shares and services using the pool. Just made sense to do it long hand.

Thank you for the prompt answers and for giving me the confidence to get on with it.

Cheers
Tony
